tests/testthat/test-linear_pool.R

```diff
@@ -453,13 +657,222 @@ test_that("samples only collected and re-indexed for simplest case", {
 })
 
 
+test_that("remainder samples are properly distributed when component models don't all forecast for every location", {
+  # There are four models, "a", "b", "c", and "d".
+  # The first three have samples for locations "222" and "888", while model "d" has samples for only horizon "222".
+  # Since not all models forecast for every unique combination of compound task id set variables,
+  # the requested output samples will be split differently across those unique combinations
+  # We want to ensure that the correct number of output samples are returned
+  # and the component models they originate from are as expected
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
+                                horizon = 1, #week
+                                target = "inc death",
+                                target_date = as.Date("2021-12-25"),
+                                output_type = "sample",
+                                output_type_id = 1:3,
+                                value = NA_real_) |>
+    dplyr::filter(model_id %in% letters[1:3] | location == "222")
+
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 1] <-
+    c(40, 30, 45, 80)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 2] <-
+    c(60, 40, 75, 20)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 3] <-
+    c(10, 70, 15, 50)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500)
+
+  set.seed(1234)
+  models_to_resample <- sample(x = letters[1:4], size = 6 %% 4)
+  expected_outputs <- expand.grid(
+    stringsAsFactors = FALSE,
+    KEEP.OUT.ATTRS = FALSE,
+    location = c("222", "888"),
+    target = "inc death",
+    target_date = as.Date("2021-12-25"),
+    component_model = letters[1:6],
+    distinct_ids = 1
+  )
+  expected_outputs$component_model[expected_outputs$location == "222" &
+                                     expected_outputs$component_model %in% letters[5:6]] <- models_to_resample
+  expected_outputs$component_model[expected_outputs$location == "888" &
+                                     expected_outputs$component_model %in% letters[4:6]] <- letters[1:3]
+  expected_outputs <- expected_outputs |>
+    dplyr::arrange(dplyr::across(c("location", "target", "target_date", "component_model"))) |>
+    dplyr::tibble()
```

Comment:
In this test and the next, I am wondering if rather than recreating the random sampling of models to create `expected_outputs`, it would make sense to test expected attributes of the actual linear pool outputs like:
 - the total number of samples adds up to the `n_output_samples`
 - within each combination of target/location/target_date,
    - for location 888, each individual model appears either 1 or 2 times
    - for location 222, models a, b, c each appear 2 times